polynote-server/src/main/scala/polynote/server/NotebookSession.scala

```diff
@@ -117,8 +117,37 @@ class NotebookSession(subscriber: KernelSubscriber, streamingHandles: StreamingH
         ZIO.foreach_(presence.flatMap(_._2.toList).map(sel => KernelStatus(sel)))(PublishMessage)
   }
 
+  // Send the results, along with some progress updates based on how many results there are
+  private def sendCellResults(results: List[CellResult], taskInfo: TaskInfo): ZIO[PublishMessage, Throwable, Unit] = {
+    val sendResults = ZIO.when(results.nonEmpty) {
+      (results.size / 10) match {
+        case 0 => ZIO.foreach_(results)(PublishMessage)
+        case chunkSize =>
+          val chunks = results.sliding(chunkSize, chunkSize).toList
+          val messages = chunks.head ::: chunks.tail.zipWithIndex.flatMap {
+            case (chunk, index) => KernelStatus(UpdatedTasks(List(taskInfo.progress(0.5 + 0.05 * index)))) :: chunk
```

Comment:
just a thought for the future, it'd be nice if `taskInfo` had an api for setting progress that could do the math for us, i.e., we could tell it how many increments there are left and it would handle knowing what current progress is and how much to add, so we don't have to remember that progress had been set to `0.5` etc etc. 

again, not part of this PR really

Reply:
Yeah, I think that would probably be a `CurrentTask` API though, and I didn't really want to go through the task manager for this.